src/main/kotlin/org/jitsi/jicofo/CodecConfig.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ 2020 - present 8x8, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.jicofo
+
+import org.jitsi.config.JitsiConfig
+import org.jitsi.metaconfig.config
+
+class VideoCodecConfig(
+    legacyBase: String,
+    newBase: String,
+    private val codecName: String
+) {
+    private val enabled: Boolean by config {
+        "$legacyBase.ENABLE_$codecName".from(JitsiConfig.legacyConfig)
+        "$newBase.enabled".from(JitsiConfig.newConfig)
+    }
+
+    fun enabled() = enabled && pt > 0
+
+    private val pt: Int by config {
+        "$legacyBase.${codecName}_PT".from(JitsiConfig.legacyConfig)
+        "$newBase.pt".from(JitsiConfig.newConfig)
+    }
+
+    @Throws(IllegalStateException::class)
+    fun pt(): Int = if (enabled()) pt else throw IllegalStateException("$codecName is not enabled.")
```

Comment:
Metaconfig has you covered for these :) You can change pt's definition to:
```
private val pt: Int by config {
    onlyIf("$codecName is enabled", ::enabled) {
        "$legacyBase.${codecName}_PT".from(JitsiConfig.legacyConfig)
        "$newBase.pt".from(JitsiConfig.newConfig)        
    }
}
```

Reply:
Not quite, because of the way legacy config disables the codec. I want `pt()` to throw an exception if the `pt` property is set to -1 (which causes the `enabled()` function to return false, while the `enabled` property might be `true`.

Reply:
Yeah I meant to refer to use the function `enabled()` in that check, not the property--I hadn't noticed the name would be ambiguous.

Reply:
The thing is `enabled()` uses the `pt` property: `fun enabled() = enabled && pt > 0`

Reply:
I think I got lost here, but whichever one you want to use you can use.  I don't know if `::enabled` will resolve to the property or the function, but if it resolves to the property you can do this to use the function:
```
private val pt: Int by config {
    onlyIf("$codecName is enabled", { enabled() }) {
        "$legacyBase.${codecName}_PT".from(JitsiConfig.legacyConfig)
        "$newBase.pt".from(JitsiConfig.newConfig)        
    }
}
```

Reply:
The problem is that the property is enabled if `enabled()` returns true (so presumably accessing the property it calls the function at some point), but the call to `enabled()` accesses the property.

Reply:
Ah ha, I see now.  👍